source/fuzz/transformation_replace_boolean_constant_with_constant_binary.cpp

```diff
@@ -0,0 +1,284 @@
+  // separately.  In each case the logic is very similar.
+  if (lhs_constant->AsFloatConstant()) {
+    assert(rhs_constant->AsFloatConstant() &&
+           "Both constants should be of the same type.");
+    if (lhs_constant->type()->AsFloat()->width() == 32) {
+      if (!float_binop_evaluates_to(lhs_constant->GetFloat(),
+                                    rhs_constant->GetFloat(),
+                                    (SpvOp)message.opcode(), expected_result)) {
+        return false;
+      }
+    } else {
+      assert(lhs_constant->type()->AsFloat()->width() == 64);
+      if (!float_binop_evaluates_to(lhs_constant->GetDouble(),
+                                    rhs_constant->GetDouble(),
+                                    (SpvOp)message.opcode(), expected_result)) {
+        return false;
+      }
+    }
+  } else {
+    assert(lhs_constant->AsIntConstant() && "Constants should be in or float.");
+    assert(rhs_constant->AsIntConstant() &&
+           "Both constants should be of the same type.");
+    if (lhs_constant->type()->AsInteger()->IsSigned()) {
+      if (lhs_constant->type()->AsInteger()->width() == 32) {
+        if (!signed_int_binop_evaluates_to(
+                lhs_constant->GetS32(), rhs_constant->GetS32(),
+                (SpvOp)message.opcode(), expected_result)) {
+          return false;
+        }
+      } else {
+        assert(lhs_constant->type()->AsInteger()->width() == 64);
+        if (!signed_int_binop_evaluates_to(
+                lhs_constant->GetS64(), rhs_constant->GetS64(),
+                (SpvOp)message.opcode(), expected_result)) {
+          return false;
+        }
+      }
+    } else {
+      if (lhs_constant->type()->AsInteger()->width() == 32) {
+        if (!unsigned_int_binop_evaluates_to(
+                lhs_constant->GetU32(), rhs_constant->GetU32(),
+                (SpvOp)message.opcode(), expected_result)) {
+          return false;
+        }
+      } else {
+        assert(lhs_constant->type()->AsInteger()->width() == 64);
+        if (!unsigned_int_binop_evaluates_to(
+                lhs_constant->GetU64(), rhs_constant->GetU64(),
+                (SpvOp)message.opcode(), expected_result)) {
+          return false;
+        }
+      }
+    }
+  }
+
+  // The id use descriptor must identify some instruction
+  return transformation::FindInstruction(message.id_use_descriptor(),
+                                         context) != nullptr;
+}
+
+opt::Instruction* Apply(
+    const protobufs::TransformationReplaceBooleanConstantWithConstantBinary&
+        message,
+    spvtools::opt::IRContext* context,
+    spvtools::fuzz::FactManager* /*unused*/) {
+  opt::analysis::Bool bool_type;
+  opt::Instruction::OperandList operands = {
+      {SPV_OPERAND_TYPE_ID, {message.lhs_id()}},
+      {SPV_OPERAND_TYPE_ID, {message.rhs_id()}}};
+  auto binary_instruction = MakeUnique<opt::Instruction>(
+      context, (SpvOp)message.opcode(),
```

Comment:
nit: static_cast<SpvOp>(message.opcode())

Reply:
Fixed all of these, thanks.